src/main/java/com/intel/gkl/smithwaterman/IntelSmithWaterman.java

```diff
@@ -30,6 +30,8 @@
     private String nativeLibraryName = "gkl_smithwaterman";
     private static boolean initialized = false;
     private IntelGKLUtils gklUtils = new IntelGKLUtils();
+    private final int MAX_SEQUENCE_LENGTH = 32*1024-1; // 2^15 - 1
+    private final int MAXIMUM_MATCH_VALUE = 64*1024; // 2^16
 
```

Comment:
good to add note here why these values, and for match values, but I think we should be adding checks where we compute the matrix score, th8s?